domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/genericlayout/GenericLayoutEntityOrderValidatorImpl.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.genericlayout;
+
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.GenericLayoutEntity;
+
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class GenericLayoutEntityOrderValidatorImpl implements GenericLayoutValidator {
+
+    private static final String ERROR_MESSAGE_INVALID_NUMBER_WITH_CASE_FIELD =
+        "DisplayOrder '%d' needs to be a valid integer for row with label '%s', case field '%s'";
+
+    @Override
+    public ValidationResult validate(List<GenericLayoutEntity> genericLayoutEntities) {
+        final ValidationResult validationResult = new ValidationResult();
+
+        for (GenericLayoutEntity entity : genericLayoutEntities) {
```

Comment:
this looping logic should have been written once and for all validators in LayoutServiceImpl.java , rather than duplicated in each validator. And a validator should only validate 1 single field.

Reply:
Agree to this one. We write entity validators and each validates validate an entity.We may need to send all entities to the validator method for validating duplicity etc..